### PR TITLE
CT-3699 move smoke test into k8s cron-jobs and remove trim-anon job from staging

### PR DESCRIPTION
**Pull request by ymao2**
## Description

The related JIRA ticket has more information for the background and context
 
This PR aims to tidy up the cron jobs  running on different envs, what I did are below
- Move the Smoke test into k8s cronjob to be inline with other jobs for better control and visibility and remove all those related gem and files ,  but I still keep the files although I commented them all just for certain period time
- Remove run-pg.sh entirely as those steps won't be run properly as it is part of building docker image and they won't get proper environment variables they need, even they can get, those actions within this file doesn't need any more 
- Remove the job of sanitising email and creating test users from staging env
- Change the job of trimming data to trim the unassigned cases only in order to keep the testings easier on staging env

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3699

### CODE REVIEW

File: config/schedule.rb [hunk 1]
```diff
@@ -4,11 +4,11 @@
 # http://en.wikipedia.org/wiki/Cron
 # Learn more: http://github.com/javan/whenever
 #
-set :output, 'log/schedule.log'
+# set :output, 'log/schedule.log'
```
Comment (willmcb) [inline config/schedule.rb:7]: probably best to remove these if they aren't needed rather than comment them out 👍

File: docker/run-pq.sh [hunk 2]
```diff
@@ -1,56 +1,56 @@
...
-
-else
-    echo "Slave server: skipping DB and cron configuration"     #| tee -a /usr/src/app/log/unicorn.log
-fi
-
-# Increment restart metric
-
-# echo "Starting unicorn"                                         | tee -a /usr/src/app/log/unicorn.log
-# exec bundle exec unicorn -p 8080 -c config/unicorn.rb 2>&1      | tee -a /usr/src/app/log/unicorn.log
+# cd /usr/src/app
```
Comment (willmcb) [inline docker/run-pq.sh:2]: I think just delete the file if its not in use anymore 👍

File: Gemfile.lock [hunk 3]
```diff
@@ -70,7 +70,7 @@ GEM
     bcrypt (3.1.16)
     bootstrap-will_paginate (1.0.0)
       will_paginate
-    brakeman (5.0.1)
+    brakeman (5.0.4)
```
Comment (willmcb) [inline Gemfile.lock:73]: Looks like there are a lot of sub dependency changes in this PR - can't really work out why they there are so many as you only changed one direct dependency (`whenever`). May be worth having a separate PR with these in so they are more easily accessible in the commit history, and so there is less scope for any issues if something goes wrong when we merge this PR :)

File: deploy.sh [hunk 4]
```diff
@@ -139,17 +139,23 @@ function _deploy() {
     kubectl set image -f k8s-deploy/${environment}/nightly_import_cronjob.yaml \
             nightly-import=${docker_image_tag} \
             --local --output yaml | kubectl apply -n $namespace -f -
+
+    # TODO: Not sure the following is useful on staging env as the smoke test runs against live url, 
+    #  but maybe it is useful to testing whether it works on staging
+    kubectl set image -f k8s-deploy/${environment}/smoke_test_cronjob.yaml \
```
Comment (willmcb) [inline deploy.sh:145]: Makes sense to me 👍

File: lib/rake_task_helpers/staging_sync.rb [hunk 5]
```diff
@@ -1,3 +1,8 @@
+# This task is mainly for trimming down size of DB and anonymising the users' information
+# for non-staging env. Not sure why its name contains 'Sync'? which cause confusing as
+# there is no step of cloning live db in this part.
+# Right now it is designed to run on daily baisis which I couldn't see the point
```
Comment (willmcb) [inline lib/rake_task_helpers/staging_sync.rb:4]: I agree I think it could run over a longer period and that would be fine. I think we should amend this comment though when this is changed as it could become confusing in future as it refers to the current point in time but this will become out of data as a statement as things change.

File: lib/smoke_test/base.rb [hunk 6]
```diff
@@ -13,6 +13,8 @@ class Base
     def self.from_env
       raise 'TEST_USER & TEST_USER_PASS env variables must be set to run smoke tests' unless ENV['TEST_USER_PASS'] && ENV['TEST_USER']
 
+      # TODO: because the smoke test currently run against the live url, it doesn't make sense to create such job on staging env
+      # it should be changed to use staging url
```
Comment (willmcb) [inline lib/smoke_test/base.rb:17]: Should we have a ticket for this then? 🤔

File: lib/tasks/staging_sync.rake [hunk 7]
```diff
@@ -4,5 +4,24 @@ namespace :db do
     task sync: :environment do
       RakeTaskHelpers::StagingSync.new.run!
     end
+
+    # as live new questions will be imported daily and current design of UI will make the action
+    # of loading pagge slower and slower , this task is to control size of the DB
+    # ideally we should change the design of UI.
```
Comment (willmcb) [inline lib/tasks/staging_sync.rake:10]: Might be worth adding a ticket for this as well as a bit of tech debt to fix.

**Review by willmcb — APPROVED** [review]
LGTM - few small non-blocking comments 👍

File: lib/rake_task_helpers/staging_sync.rb (hunk 5 above)
Comment (ymao2) [reply lib/rake_task_helpers/staging_sync.rb:4]: I think I just remove those comments.

File: lib/smoke_test/base.rb (hunk 6 above)
Comment (ymao2) [reply lib/smoke_test/base.rb:17]: I am going to remove this comment,  had a brief discussion with Ian, it seems like the smoke test is designed like this

File: Gemfile.lock (hunk 3 above)
Comment (ymao2) [reply Gemfile.lock]: I think it is because I ran bundle update,  now I  narrowed down by running `bundle update whenever`, now it looks better
